Modules/_ctypes/_ctypes.c

```diff
@@ -967,18 +967,30 @@ CDataType_in_dll_impl(PyObject *type, PyTypeObject *cls, PyObject *dll,
         return NULL;
     }
 #else
+    dlerror();
     address = (void *)dlsym(handle, name);
-    if (!address) {
 #ifdef __CYGWIN__
-/* dlerror() isn't very helpful on cygwin */
+    if (!address) {
+        /* dlerror() isn't very helpful on cygwin */
         PyErr_Format(PyExc_ValueError,
                      "symbol '%s' not found",
                      name);
+        return NULL;
+    }
 #else
-        PyErr_SetString(PyExc_ValueError, dlerror());
-#endif
+    char *dlerr;
+    dlerr = dlerror();
+    if (dlerr) {
+        PyErr_SetString(PyExc_ValueError, dlerr);
```

Comment:
I'm worried that there was a second bug here: this assumes that UTF-8 is the default locale. I'm pretty sure that can cause problems, so I *think* we're going to want to use `PyUnicode_DecodeLocale` here, and then pass the result to `PyErr_SetObject`.

Reply:
Should this be a separate issue/investigation, followed by another PR?

In the sense that if we make a decision on this, we'll have to apply throughout.

Reply:
I'll investigate this a little further. It's probably better to deal with it in another PR. For now, could you add an `// XXX Locale problems?` or something similar?